interfaces/builtin/desktop.go

```diff
@@ -287,6 +287,14 @@ dbus (receive)
     member=ActiveChanged
     peer=(label=unconfined),
 
+# Allow to get the current idle time only from Mutter
+dbus (send)
+    bus=session
+    path="/org/gnome/Mutter/IdleMonitor/Core"
+    interface="org.gnome.Mutter.IdleMonitor"
+    member="GetIdletime"
+    peer=(label=unconfined),
+
```

Comment:
This seems like something that'd make sense to include in `desktopConnectedPlugAppArmor` rather than `desktopConnectedPlugAppArmorClassic`. We'd need to use something other than `label=unconfined` when it's not an implicit slot.

We can always handle this later if you'd prefer though.

Reply:
Yes, that is a good point. So ideally we want this in `desktopConnectedPlugAppArmor` but then to handle the `label=unconfined` case properly - the only snap which provides this DBus name currently is the ubuntu-desktop-session snap - in which case the label would be `snap.ubuntu-desktop-session` right? - I'll see what I can come up with to handle that properly in this PR.

Reply:
Or we just drop the `peer=(label=...)` restriction on this rule entirely / switch it to `peer=(name=org.gnome.Mutter.IdleMonitor),` instead

Reply:
I went with the option to change this to `peer=(name=org.gnome.Mutter.IdleMonitor),` instead - let me know what you think.

Reply:
This is closer to what I was thinking of (although probably using `implicitSystemConnectedSlot(slot)` rather than `release.OnClassic`):

https://github.com/snapcore/snapd/blob/6d1a3ceabcf3618c0c931f21391ffb198cdc52c3/interfaces/builtin/network_manager.go#L522-L532

It's a little more involved than your current patch, since we haven't yet got the boilerplate for this in the interface. That's why I said I'd understand if you left it as something to deal with in future.

Reply:
Yeah I had assumed you meant something like this - I also prefer this approach to be honest as it ends up being more strictly correct than my current proposal. Will try and rework it along these lines. Thanks @jhenstridge.

Reply:
Hopefully [b56a612](https://github.com/snapcore/snapd/pull/13304/commits/b56a61236ec6ebf2da71ad9666f32c833e5d48db) is what you had in mind.